Mathlib/RingTheory/FinitePresentation.lean

```diff
@@ -450,6 +450,57 @@ end FinitePresentation
 
 end RingHom
 
+namespace RingHom.FinitePresentation
+universe u v
+variable {R : Type u} [CommRing R]
+
+open Polynomial
+
+/-- Induction principle for finitely presented ring homomorphisms.
+
+For a property to hold for all finitely presented ring homs, it suffices for it to hold for
+`Polynomial.C : R → R[X]`, surjective ring homs with finitely generated kernels, and to be closed
+under composition. -/
```

Comment:
Can you more clearly explain the role of `P` and `Q` in this docstring? For example, you assume `P` holds for `Polynomial.C` but the conclusion is for `Q`, which doesn't line up with the prose here.

Reply:
Morally P=Q but because of universe issues you need two of them.